src/wallet/spend.cpp

```diff
@@ -563,12 +563,15 @@ util::Result<SelectionResult> ChooseSelectionResult(const CAmount& nTargetValue,
         }
     };
 
+    // Maximum allowed transaction weight
+    int max_weight = MAX_STANDARD_TX_WEIGHT - (coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR);
```

Comment:
it's the maximum allowed transaction weight, not only for the inputs.

Reply:
> it's the maximum allowed transaction weight, not only for the inputs.

I don't think so? The coin selection algos don't have any knowledge about the additional data of the to-be-created tx (static header size + outputs), so they can only work with a weight limit on the inputs.

Reply:
I’m confused, @aureleoules’s suggestion makes sense to me. If it’s the maximum allowed transaction weight, why is it reduced by the `tx_noinputs_size`? If it’s supposed to be the allowance for inputs, it might additionally need to be reduced by an allowance for the change output?

Reply:
> it's the maximum allowed transaction weight, not only for the inputs.
I don't think so? The coin selection algos don't have any knowledge about the additional data of the to-be-created tx (static header size + outputs), so they can only work with a weight limit on the inputs.

oops, I shouldn't answer comments at night. My bad.
it's ok to say that is the max inputs weight. Will rename the variable in the next push.

> If it’s supposed to be the allowance for inputs, it might additionally need to be reduced by an allowance for the change output?

Good point. Need to also deduce the `change_spend_size` for knapsack and SRD.

Reply:
pushed